Packages/com.unity.inputsystem/InputSystem/Plugins/Users/InputUser.cs

```diff
@@ -1506,10 +1506,17 @@ private static void OnDeviceChange(InputDevice device, InputDeviceChange change)
                 // New device was added. See if it was a device we previously lost on a user.
                 case InputDeviceChange.Added:
                 {
-                    // Could be a previously lost device. Could affect multiple users. Repeatedly search in
-                    // s_AllLostDevices until we can't find the device anymore.
-                    var deviceIndex = s_GlobalState.allLostDevices.IndexOfReference(device, s_GlobalState.allLostDeviceCount);
-                    while (deviceIndex != -1)
+                    int FindLostDevice(int deviceIndex, int deviceId)
+                    {   // Note that we compare by Device ID since the lost device may be another instance
+                        // since its recreated during layout override. For other cases its the same reference.
```

Comment:
I think this needs to compare *both* by reference and by device ID. We may also have the situation that the *same* `InputDevice` instance is coming back with a *new* device ID (happens when a device is retained on the disconnected list and then pulled back out when the native backend reports the same device under a new ID).

Reply:
Probably intended to use `deviceId` argument.

Realize now that the anonymous function at line 1512 also creates garbage. Don't know... maybe it doesn't matter on this path.